tests/test_core.py

```diff
@@ -143,13 +167,42 @@ def test_scheme_all(self):
 
             options = obj['options']
             self.assertIsInstance(options, dict)
-            channel, psk, ssid = options['wireless-channel'], options['wpa-psk'], options['wpa-ssid']
-            self.assertIsInstance(channel, str)
-            self.assertIsInstance(psk, str)
-            self.assertIsInstance(ssid, str)
-            self.assertTrue(len(channel) > 0)
-            self.assertTrue(len(psk) > 0)
-            self.assertTrue(len(ssid) > 0)
+            channel, psk, ssid = options.get('wireless-channel'), options.get('wpa-psk'), options.get('wpa-ssid')
+            if channel is not None:
```

Comment:
Some things to comment here:
- What is it exactly that you want to test? E.g, if channel is not defined, should the test fail or not? 
- Tests should not be printing much stuff to output, but rather passing quietly or failing with error messages.
- Avoid re-doing the same questions.